### ISSUE TITLE
@Assert\NotNull annotation automatically added in all fields

### ISSUE THREAD

**Issue by Simperfit** [api-platform/schema-generator#107]
Taken from api-platform/api-platform#363

### Actual Result
When generate entity file using schema.yml file, `@Assert\NotNull` annotation automatically added.

### Expected Result
`@Assert\NotNull` should not add in entity file

#### /api-platform/app/config/schema.yml

```
    Users:                       
        parent: false
        properties:
            createdAt: { range: "DateTime" }
            updatedAt: { range: "DateTime" } 
            username: { range: "Text", ormColumn: 'type="string", length=255' }
            email: { range: "Text", ormColumn: 'type="string", length=255' }
            password: { range: "Text", ormColumn: 'type="string", length=255' }
```

#### Entity generate using command:

```
php vendor/bin/schema generate-types src/ app/config/schema.yml
```

#### /api-platform/src/AppBundle/Entity/Users.php

```
<?php

namespace AppBundle\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * The most generic type of item.
 *
 * @see http://schema.org/Thing Documentation on Schema.org
 *
 * @ORM\Entity
 * @ApiResource(iri="http://schema.org/Thing")
 */
class Users
{
    /**
     * @var int
     *
     * @ORM\Column(type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var \DateTime
     *
     * @ORM\Column(type="datetime")
     * @Assert\DateTime
     * @Assert\NotNull
     */
    private $createdAt;

    /**
     * @var \DateTime
     *
     * @ORM\Column(type="datetime")
     * @Assert\DateTime
     * @Assert\NotNull
     */
    private $updatedAt;

    /**
     * @var string
     *
     * @ORM\Column(type="string", length=255)
     * @Assert\Type(type="string")
     * @Assert\NotNull
     */
    private $username;

    /**
     * @var string
     *
     * @ORM\Column(type="string", length=255)
     * @Assert\Type(type="string")
     * @Assert\NotNull
     */
    private $email;

    /**
     * @var string
     *
     * @ORM\Column(type="string", length=255)
     * @Assert\Type(type="string")
     * @Assert\NotNull
     */
    private $password;
    ....
    ....
    ....
}
```

**Comment by Charl13** [comment]
Same issue.

```
namespaces:
  entity: App\Model
types:
  User:
    properties:
      email: { nullable: true, range: "Text" }
      authenticationToken: { nullable: true, range: "Text" }
      awsAccessToken: { nullable: true, range: "Text" }
      awsRefreshToken: { nullable: true, range: "Text" }
      createdOn: { nullable: true, range: "DateTime" }
      updatedOn: { nullable: true, range: "DateTime" }
```

```
<?php

declare(strict_types=1);

namespace App\Model;

use ApiPlatform\Core\Annotation\ApiResource;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * The most generic type of item.
 *
 * @see http://schema.org/Thing Documentation on Schema.org
 *
 * @author Charlie Vieillard <charlie@labrooms.com>
 *
 * @ORM\Entity
 * @ApiResource(iri="http://schema.org/Thing")
 */
class User
{
    /**
     * @var int|null
     *
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(type="text")
     * @Assert\NotNull
     */
    private $email;

    /**
     * @var string
     *
     * @ORM\Column(type="text")
     * @Assert\NotNull
     */
    private $authenticationToken;

    /**
     * @var string
     *
     * @ORM\Column(type="text")
     * @Assert\NotNull
     */
    private $awsAccessToken;

    /**
     * @var string
     *
     * @ORM\Column(type="text")
     * @Assert\NotNull
     */
    private $awsRefreshToken;

    /**
     * @var \DateTimeInterface
     *
     * @ORM\Column(type="datetime")
     * @Assert\DateTime
     * @Assert\NotNull
     */
    private $createdOn;

    /**
     * @var \DateTimeInterface
     *
     * @ORM\Column(type="datetime")
     * @Assert\DateTime
     * @Assert\NotNull
     */
    private $updatedOn;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setEmail(string $email): void
    {
        $this->email = $email;
    }

    public function getEmail(): string
    {
        return $this->email;
    }

    public function setAuthenticationToken(string $authenticationToken): void
    {
        $this->authenticationToken = $authenticationToken;
    }

    public function getAuthenticationToken(): string
    {
        return $this->authenticationToken;
    }

    public function setAwsAccessToken(string $awsAccessToken): void
    {
        $this->awsAccessToken = $awsAccessToken;
    }

    public function getAwsAccessToken(): string
    {
        return $this->awsAccessToken;
    }

    public function setAwsRefreshToken(string $awsRefreshToken): void
    {
        $this->awsRefreshToken = $awsRefreshToken;
    }

    public function getAwsRefreshToken(): string
    {
        return $this->awsRefreshToken;
    }

    public function setCreatedOn(\DateTimeInterface $createdOn): void
    {
        $this->createdOn = $createdOn;
    }

    public function getCreatedOn(): \DateTimeInterface
    {
        return $this->createdOn;
    }

    public function setUpdatedOn(\DateTimeInterface $updatedOn): void
    {
        $this->updatedOn = $updatedOn;
    }

    public function getUpdatedOn(): \DateTimeInterface
    {
        return $this->updatedOn;
    }
}
```

**Comment by soyuka** [comment]
https://github.com/api-platform/schema-generator/commit/c8d2a7fbfe8d83f613b583ca78a09b32690ac87d

**Comment by Charl13** [comment]
@soyuka How is this relevant?

**Comment by banandam** [comment]
@soyuka Now also i face same issue.

**Comment by dunglas** [comment]
Closed by #160
